content/articles/signal-handling-in-c++/index.md

```diff
@@ -0,0 +1,141 @@
+---
+layout: engineering-education
+status: publish
+published: true
+url: /signal-handling-in-cpp/
+title: Signal Handling In C++
+description: This article will provide an overview of signals and how they can be handled in C++.
+author: felix-maina
+date: 2021-12-16T00:00:00-13:30
+topics: [Security]
+excerpt_separator: <!--more-->
+images:
+  - url: /engineering-education/signal-handling-in-cpp/hero.jpg
+    alt: Signal Handling In C++
+---
+There are interruptions that come up in your program, which are delivered to a process that runs on an operating system and pauses the running process. These interrupts are called signals.
+<!--more-->
+The signals can be generated by pressing `Ctrl+C` on various operating systems (Windows, Mac OS X, LINUX, or UNIX).
+
+When using the C++ programming language, the interrupts can be handled in the operating systems mentioned above.
+
+But before we go into signal handling in C++, you may want to know a little about the programming language.
+
+### What is C++ programming language?
+C++ is a general-purpose cross-platform programming computer language, which is case-sensitive and free-form.
+
+The language is highly flexible and scalable and gives programmers an optimized level of control over system memory and resources. It is used for the development of operating systems (OS), browsers, apps, games, software, graphical user interfaces (GUIs), etc.
+
+The imperative and compiled language is a powerful framework that supports different methods of programming such as object-oriented paradigm, functional programming, generic programming, and procedural programming.
+
+### Signal handling in C++
+We understand that signals are interrupts that stop an OS in its task to attend to another task, and to stop these actions, the signals need to be handled.
+
+In C++, signals are provided by the language. It catch the interrupting signals and solve the problem that causes the program to pause while executing its task.
+
+Below is the C++ signal library, signals that a developer can work with and their operations.
+
+#### Signal operations
+1. `SIGINT`(Signal Interrupt) - This signal is used to generate an interactive attention signal as a "program interrupt" at the receiver end. When the OS generates an interrupt, this C++ signal interrupts the interrupt and allows the ongoing task to continue.
+<!--  -->
```

Comment:
This statement: ` ... this C++ signal interrupts the interrupt ...` is a bit confusing. Please rephrase it to remove the confusion between the word "interrupt".